### ISSUE TITLE
type naming annotation.

### ISSUE THREAD

**Issue by LuisAyuso** [insieme/insieme#136]
There is a weird behavior related to name annotations in types.
type_converter.cpp:396 & type_converter.cpp:367 attach the declaration name to the node.
When in combination with C anonymous types, the annotation is ignored. This was being done in the annotation mechanism and a warning was prompt.

If the Insieme-mangled name is used, some outputed codes fail to be compiled by the backend compiler. 

Currently annonymous types are not annotated, which leads me to the question: what is this annotation good for? it seems that the backend does some naming when available, and it is bypassed when missing.

In any case, since the magled name is used to create the tag-type, this annotation seems redundant and should be removed from both frontend and backend.

**Comment by PeterTh** [comment]
I agree. In fact, it should be investigated if at this point we can remove name annotations from everything except variables, since both types and lambdas have names (tags) anyway.
